search/search_repository.go

```diff
@@ -263,6 +264,18 @@ func parseMap(queryMap map[string]interface{}, q *Query) {
 		case bool:
 			s := concreteVal
 			q.Negate = s
+		case map[string]interface{}:
+			if v, ok := concreteVal["$IN"]; ok {
+				q.Name = Q_OR
+				c := &q.Children
+				for _, vl := range v.([]interface{}) {
```

Comment:
Weird! This amount of castings will be a source of panic errors :/. I am not sure if we should add a defer panic to this function to capture them.